src/definitions/modules/calendar.js

```diff
@@ -885,7 +885,7 @@
                         var endDate = module.get.endDate();
                         if (!!endDate && !!date && date > endDate) {
                             // selected date is greater than end date in range, so clear end date
-                            module.set.endDate(undefined);
+                            module.set.endDate();
```

Comment:
as discussed in https://github.com/fomantic/Fomantic-UI/pull/2584#discussion_r1039529752